### PR TITLE
ninja requires this policy to operate

### PR DESCRIPTION
**Pull request by PDoakORNL**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,11 @@ IF(COMMAND cmake_policy)
     # are used in try_compile test cmake projects.
     # CHECK_CXX_SOURCE_COMPILES and others depend on try_compile
   endif(CMAKE_VERSION VERSION_GREATER 3.1.3 AND NOT $ENV{CRAYPE_VERSION} MATCHES ".")
+  IF(CMAKE_GENERATOR MATCHES "Ninja")
+    cmake_policy(SET CMP0058 NEW)
+    # Ninja requires this for maximum efficiency
+    # Try it, its faster than make
```
Comment (ye-luo) [inline CMakeLists.txt]: Please 1) rebase on the current develop 2) remove this comment and put it in the PR description.

**Comment by ye-luo** [comment]
A general comment, it is good to have support of ninja but not necessary for its availability. Effort is more needed on cleaning up the dependency which benefit all building tools.

**Comment by PDoakORNL** [comment]
I always build with it when I'm debugging much better partial builds. I've had this in my own CMakelist.txt for awhile.

**Comment by ye-luo** [comment]
It's good to increase your productivity.

**Comment by ye-luo** [comment]
Okay to test

**Pull request merged**